lib/ansible/modules/network/frr/frr_facts.py

```diff
@@ -0,0 +1,470 @@
+
+        if ldp_supported:
+            data = self.responses[1]
+            if data:
+                self.facts['mpls_ldp_neighbors'] = self.populate_mpls_ldp_neighbors(data)
+
+    def parse_interfaces(self, data):
+        parsed = dict()
+        key = ''
+        for line in data.split('\n'):
+            if len(line) == 0:
+                continue
+            elif line[0] == ' ':
+                parsed[key] += '\n%s' % line
+            else:
+                match = re.match(r'^Interface (\S+)', line)
+                if match:
+                    key = match.group(1)
+                    parsed[key] = line
+        return parsed
+
+    def populate_interfaces(self, interfaces):
+        facts = dict()
+        counters = ['description', 'macaddress', 'type', 'vrf', 'mtu', 'bandwidth', 'lineprotocol',
+                    'operstatus']
+        for key, value in iteritems(interfaces):
+            intf = dict()
+            for x in counters:
+                meth = getattr(self, 'parse_%s' % x, None)
+                intf[x] = meth(value)
+            facts[key] = intf
+        return facts
+
+    def parse_description(self, data):
+        match = re.search(r'Description: (.+)', data)
+        if match:
+            return match.group(1)
+
+    def parse_macaddress(self, data):
+        match = re.search(r'HWaddr: (\S+)', data)
+        if match:
+            return match.group(1)
+
+    def parse_type(self, data):
+        match = re.search(r'Type: (\S+)', data)
+        if match:
+            return match.group(1)
+
+    def parse_vrf(self, data):
+        match = re.search(r'vrf: (\S+)', data)
+        if match:
+            return match.group(1)
+
+    def parse_mtu(self, data):
+        match = re.search(r'mtu (\d+)', data)
+        if match:
+            return int(match.group(1))
+
+    def parse_bandwidth(self, data):
+        match = re.search(r'bandwidth (\d+)', data)
+        if match:
+            return int(match.group(1))
+
+    def parse_lineprotocol(self, data):
+        match = re.search(r'line protocol is (\S+)', data)
+        if match:
+            return match.group(1)
+
+    def parse_operstatus(self, data):
+        match = re.search(r'^(?:.+) is (.+),', data, re.M)
+        if match:
+            return match.group(1)
+
+    def populate_ipv4_interfaces(self, data):
+        for key, value in data.items():
+            self.facts['interfaces'][key]['ipv4'] = list()
+            primary_address = addresses = []
+            primary_address = re.findall(r'inet (\S+) broadcast (?:\S+)(?:\s{2,})', value, re.M)
+            addresses = re.findall(r'inet (\S+) broadcast (?:\S+)(?:\s+)secondary', value, re.M)
+            if len(primary_address) == 0:
+                continue
+            addresses.append(primary_address[0])
+            for address in addresses:
+                addr, subnet = address.split("/")
+                ipv4 = dict(address=addr.strip(), subnet=subnet.strip())
```

Comment:
This logic seems to extract the ip address and subnet if that's the case please check if the utility functions [here](https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/common/network.py) can be reused.

Reply:
@ganeshrn Could not find a matching method that could be used in this scenario.